File: bookstore-backend/src/main/java/com/example/bookstore_db/dto/PaymentUrlResponse.java
package com.example.bookstore_db.dto;

/**
 * Response khi tạo đơn hàng.
 * COD: payUrl + deeplink = null
 * MOMO: payUrl và deeplink từ MoMo API
 */
public class PaymentUrlResponse {
    private Long orderId;
    private String payUrl;      // URL mở trên browser (luôn có nếu MOMO)
    private String deeplink;    // URL mở thẳng app MoMo (có thể null)
    private String paymentMethod;

    public PaymentUrlResponse(Long orderId, String payUrl, String deeplink, String paymentMethod) {
        this.orderId = orderId;
        this.payUrl = payUrl;
        this.deeplink = deeplink;
        this.paymentMethod = paymentMethod;
    }

    public Long getOrderId() { return orderId; }
    public String getPayUrl() { return payUrl; }
    public String getDeeplink() { return deeplink; }
    public String getPaymentMethod() { return paymentMethod; }
}

File: BookStore/app/src/main/java/com/example/bookstore/data/dto/response/PaymentUrlResponse.kt
package com.example.bookstore.data.dto.response

data class PaymentUrlResponse(
    val orderId: Long,
    val payUrl: String?,      // URL mở browser (MoMo web)
    val deeplink: String?,    // URL mở thẳng app MoMo (có thể null)
    val paymentMethod: String = ""
)

